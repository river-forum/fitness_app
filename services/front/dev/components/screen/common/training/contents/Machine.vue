<template>
  <div class="test">
    <InputDate />
    <div class="mt-20 accordion-wrapper">
      <Accordion accordion-title="肩" :arr-each-part="arrEachPart('肩')" />
      <Accordion accordion-title="腕" :arr-each-part="arrEachPart('腕')" />
      <Accordion accordion-title="胸" :arr-each-part="arrEachPart('胸')" />
      <Accordion accordion-title="お腹" :arr-each-part="arrEachPart('お腹')" />
      <Accordion accordion-title="背中" :arr-each-part="arrEachPart('背中')" />
      <Accordion accordion-title="腰" :arr-each-part="arrEachPart('腰')" />
      <Accordion accordion-title="脚" :arr-each-part="arrEachPart('脚')" />
    </div>
  </div>
</template>

<script>
import { defineComponent } from '@nuxtjs/composition-api'
import MasterData from '@/data/master'
import Accordion from '../contents/machine/Accordion'

export default defineComponent({
  components: {
    Accordion,
  },
  setup() {
    function arrEachPart(name) {
      const partsNameArr = MasterData.filter((data) => name === data.partsName)

      return partsNameArr
    }

    return {
      arrEachPart,
    }
  },
})
</script>
