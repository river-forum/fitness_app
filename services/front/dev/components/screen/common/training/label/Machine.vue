<template>
  <Frame
    current-color="#718ec5"
    :active="active"
    @click.native="selectedListener"
  >
    <template slot="contents">マシーン</template>
  </Frame>
</template>
<script>
import { defineComponent, computed } from '@nuxtjs/composition-api'
import Frame from './_Frame'

const IDENTIFY = 'machine'

export default defineComponent({
  components: {
    Frame,
  },
  props: {
    current: {
      type: String,
      required: true,
    },
  },
  setup(props, ctx) {
    const active = computed(() => props.current === IDENTIFY)
    function selectedListener() {
      ctx.emit('selected', IDENTIFY)
    }

    return {
      selectedListener,
      active,
    }
  },
})
</script>
