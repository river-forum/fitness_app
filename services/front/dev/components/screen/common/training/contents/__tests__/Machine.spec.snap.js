import { mount } from '@vue/test-utils'
import Machine from '../Machine.vue'

it('Machine Component renders correctly', () => {
  const app = mount(Machine)

  expect(app.html()).toMatchSnapshot()
})
