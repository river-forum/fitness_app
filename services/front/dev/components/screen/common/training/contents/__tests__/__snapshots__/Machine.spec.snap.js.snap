// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Machine Component renders correctly 1`] = `
"<div class=\\"test\\">
  <inputdate></inputdate>
  <div class=\\"mt-20 accordion-wrapper\\">
    <div class=\\"px-6 accordion\\"><button type=\\"button\\" class=\\"px-2 py-4 text-left accordion__trigger\\">
        肩
      </button>
      <transition-stub name=\\"js-accordion\\">
        <!---->
      </transition-stub>
    </div>
    <div class=\\"px-6 accordion\\"><button type=\\"button\\" class=\\"px-2 py-4 text-left accordion__trigger\\">
        腕
      </button>
      <transition-stub name=\\"js-accordion\\">
        <!---->
      </transition-stub>
    </div>
    <div class=\\"px-6 accordion\\"><button type=\\"button\\" class=\\"px-2 py-4 text-left accordion__trigger\\">
        胸
      </button>
      <transition-stub name=\\"js-accordion\\">
        <!---->
      </transition-stub>
    </div>
    <div class=\\"px-6 accordion\\"><button type=\\"button\\" class=\\"px-2 py-4 text-left accordion__trigger\\">
        お腹
      </button>
      <transition-stub name=\\"js-accordion\\">
        <!---->
      </transition-stub>
    </div>
    <div class=\\"px-6 accordion\\"><button type=\\"button\\" class=\\"px-2 py-4 text-left accordion__trigger\\">
        背中
      </button>
      <transition-stub name=\\"js-accordion\\">
        <!---->
      </transition-stub>
    </div>
    <div class=\\"px-6 accordion\\"><button type=\\"button\\" class=\\"px-2 py-4 text-left accordion__trigger\\">
        腰
      </button>
      <transition-stub name=\\"js-accordion\\">
        <!---->
      </transition-stub>
    </div>
    <div class=\\"px-6 accordion\\"><button type=\\"button\\" class=\\"px-2 py-4 text-left accordion__trigger\\">
        脚
      </button>
      <transition-stub name=\\"js-accordion\\">
        <!---->
      </transition-stub>
    </div>
  </div>
</div>"
`;
